public/app/core/services/keybindingSrv.ts

```diff
@@ -175,6 +181,32 @@ export class KeybindingSrv {
     this.$location.search(search);
   }
 
+  renderSaveDashboardModal = (dashboard: DashboardModel) => {
+    // when saveDashboardModalRoot is in the DOM it means modal is visible
+    if (document.getElementById('saveDashboardModalRoot')) {
+      return;
+    }
+
+    const el = document.createElement('div');
+    el.setAttribute('id', 'saveDashboardModalRoot');
+    document.body.append(el);
+    const reactRoot = document.getElementById('saveDashboardModalRoot');
+
+    return ReactDOM.render(
+      React.createElement(SaveDashboardModalProxyAngular, {
+        component: SaveDashboardModalProxy,
+        props: {
+          dashboard,
+          onClose: () => {
+            ReactDOM.unmountComponentAtNode(reactRoot);
+            reactRoot.remove();
+          },
+        },
+      }),
+      reactRoot
+    );
+  };
```

Comment:
is there a way we can do this with an event instead like before? so we only open the modal from one place? This feels not as great as simple sending an event, and having one owner responsible for opening and rendering the modal.

Reply:
Yes, you are right. I'm gonna do that.